<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>919</width>
    <height>756</height>
   </rect>
  </property>
  <property name="acceptDrops">
   <bool>true</bool>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="toolButtonStyle">
   <enum>Qt::ToolButtonTextOnly</enum>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPushButton" name="loadfitnessButton">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>10</y>
      <width>80</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Load</string>
    </property>
   </widget>
   <widget class="PlotWidget" name="fitnessPlot">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>40</y>
      <width>301</width>
      <height>251</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>10</y>
      <width>70</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Fitness</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>360</y>
      <width>70</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Pcross</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>460</y>
      <width>70</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Pmut</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>530</y>
      <width>70</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Selection</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>560</y>
      <width>81</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Replacement</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="pcrossField">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>360</y>
      <width>113</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="pmutField">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>460</y>
      <width>113</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QSlider" name="universeSlider">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>420</y>
      <width>160</width>
      <height>40</height>
     </rect>
    </property>
    <property name="pageStep">
     <number>1</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="invertedAppearance">
     <bool>false</bool>
    </property>
    <property name="invertedControls">
     <bool>false</bool>
    </property>
    <property name="tickInterval">
     <number>0</number>
    </property>
   </widget>
   <widget class="QGraphicsView" name="universePlot">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>40</y>
      <width>371</width>
      <height>311</height>
     </rect>
    </property>
    <property name="mouseTracking">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>10</y>
      <width>70</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Universe</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>430</y>
      <width>70</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Generation</string>
    </property>
   </widget>
   <widget class="QPushButton" name="runButton">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>10</y>
      <width>41</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset>
      <normaloff>play.png</normaloff>play.png</iconset>
    </property>
   </widget>
   <widget class="QPushButton" name="pauseButton">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>10</y>
      <width>41</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset>
      <normaloff>pause.png</normaloff>pause.png</iconset>
    </property>
   </widget>
   <widget class="QPushButton" name="stopButton">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>10</y>
      <width>41</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset>
      <normaloff>stop.png</normaloff>stop.png</iconset>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>330</y>
      <width>70</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Crossover</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>420</y>
      <width>70</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Mutation</string>
    </property>
   </widget>
   <widget class="QComboBox" name="crossoverCombo">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>330</y>
      <width>111</width>
      <height>22</height>
     </rect>
    </property>
    <property name="editable">
     <bool>false</bool>
    </property>
    <property name="currentText">
     <string>Single Point</string>
    </property>
    <item>
     <property name="text">
      <string>Single Point</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Two Points</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Uniform</string>
     </property>
    </item>
   </widget>
   <widget class="QComboBox" name="mutationCombo">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>420</y>
      <width>111</width>
      <height>22</height>
     </rect>
    </property>
    <property name="editable">
     <bool>false</bool>
    </property>
    <property name="currentText">
     <string>Random</string>
    </property>
    <item>
     <property name="text">
      <string>Random</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Swap</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Inversion</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Scramble</string>
     </property>
    </item>
   </widget>
   <widget class="QComboBox" name="comboBox_3">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>530</y>
      <width>111</width>
      <height>22</height>
     </rect>
    </property>
    <property name="editable">
     <bool>false</bool>
    </property>
    <property name="currentText">
     <string>Steady State</string>
    </property>
    <item>
     <property name="text">
      <string>Steady State</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Roulette Wheel</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Stochastic Universal</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Rank</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Random</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Tournament</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>590</y>
      <width>81</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Population</string>
    </property>
   </widget>
   <widget class="QPushButton" name="exportfitnessplotButton">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>10</y>
      <width>80</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Export Plot</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_3">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>10</y>
      <width>121</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Export Simulation</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="populationField">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>590</y>
      <width>113</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>620</y>
      <width>81</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Generations</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="generationsField">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>620</y>
      <width>113</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>390</y>
      <width>81</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Num Parents</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="numparentsField">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>390</y>
      <width>113</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextBrowser" name="simulationLog">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>480</y>
      <width>371</width>
      <height>192</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>450</y>
      <width>70</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Log</string>
    </property>
   </widget>
   <widget class="QPushButton" name="exportlogButton">
    <property name="geometry">
     <rect>
      <x>670</x>
      <y>450</y>
      <width>80</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Export Log</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_5">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>450</y>
      <width>80</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Settings</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_6">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>10</y>
      <width>80</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Settings</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_14">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>300</y>
      <width>71</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Num Genes</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="numgenesField">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>300</y>
      <width>113</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_15">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>490</y>
      <width>70</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>% Mut Gen</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="percmutgeneField">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>490</y>
      <width>113</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QCheckBox" name="clearFitnessplotCheckbox">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>680</y>
      <width>221</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Clear Fitness Plot Before Sim</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <property name="tristate">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLabel" name="universeGen">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>430</y>
      <width>70</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QLabel" name="chosenChromosome">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>380</y>
      <width>491</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>[]</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_16">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>370</y>
      <width>151</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Chosen Chromosome</string>
    </property>
   </widget>
   <widget class="QLabel" name="chosenFitness">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>410</y>
      <width>70</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_17">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>410</y>
      <width>151</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Chosen Fitness</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>919</width>
     <height>19</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QGraphicsView</extends>
   <header>pyqtgraph</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
